File: app/src/main/java/com/kelsos/mbrc/events/bus/RxBus.java
package com.kelsos.mbrc.events.bus;

import rx.Subscription;
import rx.functions.Action1;

public interface RxBus {
  <T> void register(Object object, Class<T> eventClass, Action1<T> onNext);

  <T> void register(Object object, Class<T> eventClass, Action1<T> onNext, boolean main);

  void unregister(Object object);

  <T> Subscription register(Class<T> eventClass, Action1<T> onNext, boolean main);

  void post(Object event);
}

File: app/src/main/kotlin/com/kelsos/mbrc/events/bus/RxBus.kt
package com.kelsos.mbrc.events.bus

import io.reactivex.Observable
import io.reactivex.disposables.Disposable
import kotlin.reflect.KClass

interface RxBus {
  fun <T> register(receiver: Any, eventClass: Class<T>, onNext: (T) -> Unit)

  fun <T> register(receiver: Any, eventClass: Class<T>, onNext: (T) -> Unit, main: Boolean)

  fun unregister(receiver: Any)

  fun <T> register(eventClass: Class<T>, onNext: (T) -> Unit, main: Boolean): Disposable

  fun <T : Any> observe(eventClass: KClass<T>): Observable<T>

  fun post(event: Any)
}
